components/ide/jetbrains/image/status/main.go

```diff
@@ -90,7 +137,7 @@ func main() {
 		return
 	}
 
-	projectDir := wsInfo.GetCheckoutLocation()
+	projectDir := resolveProjectDir(alias, wsInfo.GetCheckoutLocation())
 	gitpodConfig, err := parseGitpodConfig(projectDir)
```

Comment:
@andreafalzetti here it is bogus, gipod config should be parsed from checkout location

I think you will need to check all usages of projectDir and understand how to use it when it is solution file.

Reply:
Maybe don't change semantic of `projectDir` but add a new variable `solutionFile`? And use it instead of `projectDir` only where it makes sense?

Reply:
@akosyakov adding new prop to launch context seems unnecessary 🤔 What about `args = append(args, resolveProjectDir(launchCtx))`, so all instances of `projectDir` are not affected but we only wrap the run script argument with the detection

Reply:
> are not affected but we only wrap the run script argument with the detection

I'm not sure that they should not be affected, for instance what is config directory for solution file?

Reply:
good catch, the settings folder actuallty changes path. I did some changes to support that 🙏